### PR TITLE
Pr/179

### PR DESCRIPTION
**Pull request by Igor-Misic**
Rebased clean version of #179

### CODE REVIEW

File: stm32f7.ld [hunk 1]
```diff
@@ -36,7 +36,7 @@
 /* Define memory regions. */
 MEMORY
 {
-	rom (rx)  : ORIGIN = 0x08000000, LENGTH = 16K
```
Comment (LorenzMeier) [inline stm32f7.ld:39]: This will break all existing F7 targets in PX4. How is this supposed to work?

**Review by LorenzMeier — CHANGES_REQUESTED** [review]
Ok, thanks!

File: stm32f7.ld (hunk 1 above)
Comment (Igor-Misic) [reply stm32f7.ld:39]: Application linker scripts are already at 0x08008000  (0x08000000 + 32K = 0x08008000)
https://github.com/PX4/PX4-Autopilot/blob/master/boards/px4/fmu-v5x/nuttx-config/scripts/script.ld#L74
image

Comment (LorenzMeier) [reply stm32f7.ld:39]: But what is the explanation for that change? What is being stored in the adjacent 16K? Flash based parameters? @davids5 ? Or was it really just a buffer that we added back in the day?

Comment (Igor-Misic) [reply stm32f7.ld:39]: With monocypher, bootloader can't fit with in 16K, and F7 is working in single-bank mode where 1 flash sector is 32K.

**Review by LorenzMeier — APPROVED** [review]

**Pull request merged**